scripts/test/Muon/MaxEntPresenter_test.py

```diff
@@ -7,140 +7,112 @@
 
 
-def test(inputs):
-    inputs["OutputPhaseTable"] = "test"
-
-
 class MaxEntPresenterTest(unittest.TestCase):
     def setUp(self):
-        self.load=mock.create_autospec(load_utils.LoadUtils,spec_set=True)
-        self.load.getCurrentWS=mock.Mock(return_value=["TEST00000001",["fwd","bkwd"]])
-        self.load.hasDataChanged = mock.MagicMock(return_value=False)
-
-        self.model=mock.create_autospec(maxent_model.MaxEntModel,spec_set=True)
-
-        self.view=mock.create_autospec(maxent_view.MaxEntView,spec_set=True)
-        #signals
-        #needed for connect in presenter
-        self.view.maxEntButtonSignal=mock.Mock()
-        self.view.cancelSignal=mock.Mock()
-        self.view.phaseSignal=mock.Mock()
+        self.context = setup_context(True)
+        self.context.data_context.instrument = 'MUSR'
+        self.context.gui_context.update({'RebinType': 'None'})
+
+        # Model
+        self.model = mock.create_autospec(maxent_model.MaxEntWrapper, spec_set=True)
+        self.model.cancel = mock.Mock()
+
+        # View
+        self.view = mock.create_autospec(maxent_view_new.MaxEntView, spec_set=True)
+        # signals
+        # needed for connect in presenter
+        self.view.maxEntButtonSignal = mock.Mock()
+        self.view.cancelSignal = mock.Mock()
         # functions
-        self.view.addItems=mock.MagicMock()
-        self.view.initMaxEntInput=mock.Mock(return_value={"InputWorkspace":"testWS","EvolChi":"out",
-                                            "ReconstructedData":"out2","ReconstructedImage":"out3","EvolAngle":"out4"})
-        self.view.deactivateCalculateButton=mock.Mock()
-        self.view.activateCalculateButton=mock.Mock()
-        self.view.usePhases = mock.Mock(return_value=False)
-         #set presenter
-        self.presenter=maxent_presenter.MaxEntPresenter(self.view,self.model,self.load)
+        self.view.addItems = mock.MagicMock()
+        self.view.deactivateCalculateButton = mock.Mock()
+        self.view.activateCalculateButton = mock.Mock()
+        self.view.input_workspace = "TEST00000001"
+        self.view.phase_table = "Construct"
+        self.view.num_points = 2048
+        self.view.inner_iterations = 10
+        self.view.outer_iterations = 10
+        self.view.double_pulse = False
+        self.view.lagrange_multiplier = 1
+        self.view.maximum_field = 1000
+        self.view.maximum_entropy_constant = 0.1
+        self.view.fit_dead_times = False
+
+        # Presenter
+        self.presenter = maxent_presenter_new.MaxEntPresenter(self.view, self.context)
 
         # make thread
-        self.thread=mock.create_autospec(thread_model.ThreadModel)
-        self.thread.start=mock.Mock()
-        self.thread.started=mock.Mock()
-        self.thread.finished=mock.Mock()
-        self.thread.setInputs=mock.Mock()
-        self.thread.loadData=mock.Mock()
+        self.thread = mock.create_autospec(thread_model.ThreadModel)
+        self.thread.start = mock.Mock()
```

Comment:
Ditto about just creating `self.thread = mock.create_autospec(...)` and getting `self.thread.started` ...etc. for free